include/revng/Pipeline/Analysis.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pipeline/Pipe.h"
+
+namespace pipeline {
+
+class Analysis {
+private:
+  PipeWrapper Pipe;
+  std::string Name;
```

Comment:
* Mmmh pipes already have a name. We're adding a layer around pipes, this leads to think that the underlying pipe can be use on its own too, while we absolutely do not want pipes to change the model.
  I expected an analysis to be similar to a pipe, more than a wrapper of a pipe.
* I'd expect an analysis to produce a model diff or something.

Reply:
we need two names. One is the name (it is the type in the yml) used to refer to a particular pipe type among the registered ones, the other is the name assigned to the analysis used to disambiguate which analysis the user wants to run. We can't use the type of the pipe because multiple analysis may have the name LLVMPipe